src/redash_stmo/integrations/iodide/extension.py

```diff
@@ -0,0 +1,62 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, you can obtain one at http://mozilla.org/MPL/2.0/.
+import requests
+import json
+import logging
+import os
+
+from redash.models import Query
+from redash.handlers.base import BaseResource, get_object_or_404
+from redash.permissions import require_permission
+
+from redash_stmo import settings
+from redash_stmo.resources import add_resource
+
+
+logger = logging.getLogger(__name__)
+
+
+class IodidePostResource(BaseResource):
+    @require_permission("view_query")
+    def get(self, query_id):
+        query = get_object_or_404(
+            Query.get_by_id,
+            query_id,
+        )
+        url = "https://stage.iodide.nonprod.dataops.mozgcp.net/api/v1/notebooks/"
+        template = open(
+            "/extension/src/redash_stmo/integrations/iodide/iodide-template.iomd",
+            "r"
+        )
+        completed_template = template.read().replace(
+            "[QUERY_ID]",
+            query_id
+        ).replace(
+            "[REDASH_API_KEY]",
+            settings.REDASH_OWN_API_KEY_FOR_IODIDE
+        ).replace(
+            "[TITLE]",
+            query.name
+        )
+        headers = {
+            "Authorization": "Token %s" % settings.IODIDE_API_KEY,
+        }
+        data = {
+            "owner": "jkarahalis@mozilla.com",
```

Comment:
This should not be hard-coded but the email address of the current user (since we're both using Mozilla auth both user account should exist, but we may want to ask users to first login on iodide). You get that with:

```python
from flask_login import current_user

data = {
    # ..
    "owner": current_user.email,
}
```

Reply:
Ideally Iodide would return a specific HTTP response code if it can't assign the notebook to a user, or return a specific error message, so we can show it to the user and tell them to first sign up on the iodide side.

Reply:
Ah, great catch. Apologies for this. This was an oversight. I don't know what Iodide does if the email address that is specified does not already have an account, but I'll look into it.

Reply:
I'll use "redash" for now, but we can change it if the Iodide folks prefer something else.